src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs

```diff
@@ -90,6 +101,12 @@ public Task SendStreamMessageWithRetryAsync(
         return SendAsyncCore(Constants.HttpClientNames.MessageResilient, api, httpMethod, new StreamItemMessage(streamId, arg), typeHint, AsAsync(handleExpectedResponse), cancellationToken);
     }
 
+    public ObjectSerializer ObjectSerializer => _payloadContentBuilder switch
+    {
+        JsonPayloadContentBuilder jsonBuilder => jsonBuilder.ObjectSerializer,
```

Comment:
This doesn't work if the `PayloadContentBuilder` is not a `JsonPayloadContentBuider`. I'd suggest injecting `IOptionsMonior<ServiceManagerOptions>` into `RestClient`, using the `ServiceManagerOptions.ObjectSerializer` or providing a default one when `ServiceManagerOptions.ObjectSerializer=null`